### ISSUE TITLE
Add an "Add AutoLoad" button to the FileSystem right-click menu for scene and script files

### ISSUE THREAD

**Issue by fire540** [godotengine/godot-proposals#3698]
### Describe the project you are working on

A 3D airplane flying game

### Describe the problem or limitation you are having in your project

Whenever you would like to add a new AutoLoad, you need to create a new script/scene, go to Project Settings -> AutoLoad, navigate to the location of the previously created script/scene, and select it from there. Navigating through the file select page to locate a file that is already visible in the FileSystem is inconvenient.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a new button to the right-click menu in the FileSystem tab (only for scene and script files) for adding it as a new AutoLoad.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

The button could be added to the top section of this menu, as that s where it fits best: 
![image](https://user-images.githubusercontent.com/67974470/146691926-7213051f-21b1-431b-91e0-0d694960893e.png)

When the button is clicked, it would bring up the Project Settings -> AutoLoad page with the "Path" and "Node Name" fields filled in, as if the file were selected through the file select window from the AutoLoad page.

![image](https://user-images.githubusercontent.com/67974470/146692021-427c3723-7879-44b3-804b-4de3fe90c503.png)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Yes, you can use the current workflow of going to Project Settings -> AutoLoad and finding the file there.

### Is there a reason why this should be core and not an add-on in the asset library?

This is a part of the editor, and it only makes a process that already exists a bit faster and more convenient.

**Comment by SilencedPerson** [comment]
yes please, i have to make at least one autoload each project and it gets old really fast.

**Comment by me2beats** [comment]
This saves only five-ten mouse clicks per a project if you have one singleton.
That's cool but at the same time
- this makes right click menu noisier (adding it as an item of "other" submenu would partially solve the problem tho)
- for users who don't need this item, the option to show it in right click menu would be useful I guess.
- this can be done as a plugin, hackily tho because plugins don't have straightforward API to add something to context menus (see https://github.com/godotengine/godot-proposals/issues/2401).

**Comment by SilencedPerson** [comment]
Five to ten seconds is long time, especially when the menu is hidden behind project settings and a tab, just navigating to the option takes that much time.

Adding one option to a list doesn't make things all that noisier or cluttered, adding them without regard for readability does that, you can have long list of options and have it not feel noisy when you respect common UI and UX conventions. Mindful use of separators and icons when needed mitigates that issue.

It's especially useful for the users who don't need this in the instance they run into that one time when they do, as it helps with discoverability immensely among other things.

Right click menu is there mostly for advanced options, you can do most of the options there by spending five to ten seconds fiddling somewhere else in the editor.

I agree plugins need easier way of adding things there, making things easier and faster is needed in general all over the editor.

**Comment by Mickeon** [comment]
I do have to agree that it introduces clutter, with it being an option not commonly used, and all. But it doesn't sound _that_ big of a deal, especially if said option is only present for scripts/scene files.

**Comment by KoBeWi** [comment]
Well, arguably setting the main scene is also uncommon action (especially if you can set it when asked before first running), but it's in the menu.

**Comment by me2beats** [comment]
> Well, arguably setting the main scene is also uncommon action (especially if you can set it when asked before first running), but it's in the menu.

For test projects this can be not so uncommon tho.
And I don't remember is there another way to change the main scene from editor.

**Comment by KoBeWi** [comment]
It can be renamed from Project Settings.
Or you can delete the current main scene and you will get prompted again on next project run.

**Comment by fire540** [comment]
I don't think would add too much clutter considering how much easier it would make adding AutoLoads. Also, some of the existing options in the right-click menu can be done in other ways too and only exist to make it a bit faster and more convenient.

**Comment by rainlizard** [comment]
The bigger issue is the singletons menu being hidden away in Project Settings. I believe it should be as prominent as the scene tree, considering it really does have such a large effect on everything. Every single one of your nodes has access to them, and yet they're hidden away, that makes no sense to me. (I even made an [addon](https://github.com/rainlizard/SingletonScriptsShortcut), though I'm positive it's not the best implementation)

If it were changed to be more conveniently accessible then this proposal might not need to exist.

**Comment by Mickeon** [comment]
>The bigger issue is the singletons menu being hidden away in Project Settings. 
> If it were changed to be more conveniently accessible then this proposal might not need to exist.

I have to agree on both. I use a similar addon that makes Autoloads/Singletons more easily accessible through a shortcut. If something like it was brought to the standard Editor and _improved_ upon to allow for easily adding new ones, I believe this proposal would become redundant.

**Comment by me2beats** [comment]
I made a plugin that works around this (adds a button to FS context menu, see the video in the repo readme)
https://github.com/me2beats/add-to-autoload

**Comment by KoBeWi** [comment]
Would https://github.com/godotengine/godot/pull/58094 resolve this?

**Comment by Chaosus** [comment]
@KoBeWi I guess only partially - a similar option could be added to the `Create scene` dialog.

**Comment by fire-forge** [comment]
@KoBeWi no, because I was proposing having the ability to add an existing script or scene as an autoload through the context menu. That PR certainly does help though.

**Comment by Chaosus** [comment]
Note: The last time I've attempted to add an option to that menu - ends up with a refusion by @KoBeWi (https://github.com/godotengine/godot/pull/55590), so maybe it's not a good idea.
